java/src/main/java/ai/onnxruntime/OrtEnvironment.java

```diff
@@ -110,6 +110,36 @@ public static synchronized OrtEnvironment getEnvironment(
     return INSTANCE;
   }
 
+  /**
+   * Creates an OrtEnvironment using the specified global thread pool options. Note unlike the other
+   * {@code getEnvironment} methods if there already is an existing OrtEnvironment this call throws
+   * {@link IllegalStateException} as we cannot guarantee that the environment has the appropriate
+   * thread pool configuration.
```

Comment:
>ation. [](start = 25, length = 6)

Still throws if already created.

Reply:
I'm not sure what you mean? As far as I can tell there's no way to check if an environment has a thread pool in it, nor what that pool looks like. I guess I could cache the options, mirror all the fields in Java and then do an equality test as part of this check?

Reply:
Disregard this comment. The behavior has changed for the better lately.

---
In reply to: [562983378](https://github.com/microsoft/onnxruntime/pull/6406#discussion_r562983378) [](ancestors = 562983378)